### PR TITLE
More fixed to TOC

### PR DESCRIPTION
**Pull request by jason-bragg**
Added commit operation to pre store actions.
This still doesn't work correctly with batching in all cases, but does not violate non-toc transaction semantics.  TOC is still not production ready.

### CODE REVIEW

**Comment by ReubenBond** [comment]
@dotnet-bot test this please

**Comment by ReubenBond** [comment]
@dotnet-bot test functional

**Comment by ReubenBond** [comment]
@dotnet-bot test functional

File: src/Orleans.Transactions/State/StorageBatch.cs [hunk 1]
```diff
@@ -256,5 +257,24 @@ public void FollowUpAction(Action action)
             }
             followUpActions.Add(action);
         }
+
+        public void AddStorePreCondition(Func<Task<bool>> action)
+        {
+            if (this.storeConditions == null)
```
Comment (xiazen) [inline src/Orleans.Transactions/State/StorageBatch.cs]: why do we lazy init this one? can we init this in the constructor?

Comment (jason-bragg) [reply src/Orleans.Transactions/State/StorageBatch.cs]: We can.  I used lazy init cause:
- Only toc uses it, so it won't be used most of the time.
- Was following the pattern used by the after actions (the 'when in rome' rule).

I don't mind changing this if you feel strongly that it should be initialized in the constructor.

Comment (xiazen) [reply src/Orleans.Transactions/State/StorageBatch.cs]: It is a small fix, I'd like for it to be fixed here. the more cleaner code is, the better for maintenance

Comment (jason-bragg) [reply src/Orleans.Transactions/State/StorageBatch.cs]: Will do, sorry I didn't get back to the feedback in time.. :/

**Pull request merged**